### ISSUE TITLE
not a valid pbkdf2_sha512 hash

### ISSUE THREAD

**Issue by maestrosound** [schoolofcode-me/price-of-chair#3]
I keep getting this error, even though I use sha passwords like '3C9909AFEC25354D551DAE21590BB26E38D53F2173B8D3DC3EEE4C047E7AB1C1EB8B85103E3BE7BA613B31BB5C9C36214DC9F14A42FD7A2FDB84856BCA5C44C2'. How can I avoid this?

**Comment by DESECTORCON** [comment]
I have the same error too!! I't seems like I have mistaken the hash password.. There is a sha512 online hasher but there isn't a pbkdf2_sha512 online hasher..

**Comment by shravansundaram** [comment]
I also have the same error. Anyone figure out the issue?

**Comment by jslvtr** [comment]
Hey guys, the issue is with the data you're saving into the database. Make sure you follow the lecture on adding the test data to the application carefully, as it adds PBKDF2_SHA512 passwords into the database.

I'll close this now as it's a course question, and not a problem with the code.

**Comment by Wray14** [comment]
> I also have the same error. Anyone figure out the issue?

I had the same problem. Let me give you an example which might help.

I had registered an email address ravi@example.com and password as 123 without using pbkdf2_sha512 encryption. Also, later I had registered test@example.com with pbkdf2_sha512 encryption.
While I tried to login, naturally I tried to login with ravi@example.com and I got this error, however when logged in with test@example.com I didn't receive any error. When I checked my db, I realised that for my first email address, password didn't get encrypted and saved as it is - 123(as I didn't use pbkdf2_sha512 encryption ) and hence it threw error. I tried to login with test@example.com where encryption was used, it verifies the encryption and doesn't throw error.

So it depends on how we save passwords into the database as Jose suggested. Hope it helps. Thank you
